src/boot/efi/ticks.c

```diff
@@ -14,36 +15,51 @@ static uint64_t ticks_read(void) {
 
         return __builtin_ia32_rdtsc();
 }
+
+static uint64_t ticks_freq_arch(void) {
+        return 0;
+}
+
 #elif defined(__aarch64__)
-static uint64_t ticks_read(void) {
+
+static uint64_t ticks_read_arch(void) {
         uint64_t val;
         asm volatile("mrs %0, cntvct_el0" : "=r"(val));
```

Comment:
btw, on llvm we probably should use `[__builtin_readcyclecounter](http://clang.llvm.org/docs/LanguageExtensions.html#builtin-readcyclecounter)` which seems to work on both arm and x86 the same

Reply:
It's clang only and returns 0 on arm, and on aarch64/x86 we want to get the frequency of the counter, which is arch-dependent. So nothing is really gained.

Also, whether the cycle counter starts from zero or is suitable for time measurements seems rather arch specific to me. I'd rather see arch-specific code for that than take a wild guess.